test/integration/middleware_test.rb

```diff
@@ -12,14 +12,36 @@ class MiddlewareIntegrationTest < ActionDispatch::IntegrationTest
       "'self'; style-src 'unsafe-inline' 'self'"
   end
 
-  test "webpack dev server adds the dev server to Content-Security-Policy" do
-    begin
+  context "webpack dev server is enabled" do
+    setup do
       Rails.configuration.webpack.dev_server.enabled = true
       Webpack::Rails::Manifest.stubs(:asset_paths).returns([])
+    end
+
+    teardown do
+      Rails.configuration.webpack.dev_server.enabled = false
+    end
+
+    test "webpack dev server adds the dev server to Content-Security-Policy" do
       visit '/'
       assert page.response_headers['Content-Security-Policy'].include?(Rails.configuration.webpack.dev_server.port.to_s)
-    ensure
-      Rails.configuration.webpack.dev_server.enabled = false
+    end
+
+    test "webpack dev server adds the dev server to Content-Security-Policy on pages triggering a welcome page" do
+      Environment.stubs(:first).returns(nil)
+      visit '/environments'
+      assert page.has_content? 'Learn more about this in the documentation.'
+      assert page.response_headers['Content-Security-Policy'].include?(Rails.configuration.webpack.dev_server.port.to_s)
+    end
+
+    context "on unauthorized page requests" do
+      def login_admin; end
+
+      test "webpack dev server adds the dev server to Content-Security-Policy as well" do
+        visit '/environments'
+        assert page.status_code, '403'
```

Comment:
`assert_equal '403', page.status_code` - this currently asserts that page.status_code is trueish